src/silx/io/utils.py

```diff
@@ -508,6 +510,18 @@ def _open_local_file(filename):
         except IOError:
             debugging_info.append((sys.exc_info(),
                                    "File '%s' can't be read as spec file." % filename))
+
+        if extension == ".fio":
+            try:
+                from . import fioh5
+                return fioh5.FioH5(filename)
+            except ImportError:
+                debugging_info.append((sys.exc_info(),
+                                       "fioh5 can't be loaded."))
+            except IOError:
+                debugging_info.append((sys.exc_info(),
+                                       "File '%s' can't be read as fio file." % filename))
```

Comment:
Sorry, I've been to fast and didn't see about that....

`fioh5` is importing `commonh5` which imports `utils` so the circular import is there.
It works because it does not import the module content but I prefer to avoid this.
I reverted the move of the import for now (see https://github.com/silx-kit/silx/pull/3539/commits/c812017cf9b7da7f4cd66a1602206e1a3064508a).
We can avoid this by e.g., having a `_utils` module which is imported in both `utils` and `commonh5` and others, but that's outside of the scope of this PR.

> fioh5.is_fiofile is now checked in fioh5.FioH5.
Great!